src/helpers/rank.ts

```diff
@@ -1,7 +1,14 @@
-import type { Claims, PropertyClaims } from '../types/claim.js'
+import { typedEntries } from '../utils/utils.js'
+import type { Claim, Claims, PropertyClaims, Rank } from '../types/claim.js'
 
 export function truthyPropertyClaims (propertyClaims: PropertyClaims): PropertyClaims {
-  const aggregate = propertyClaims.reduce(aggregatePerRank, {})
+  const aggregate: Partial<Record<Rank, Claim[]>> = {}
+  for (const claim of propertyClaims) {
+    const { rank } = claim
+    aggregate[rank] ??= []
```

Comment:
[The `??=` operator requires NodeJS >= v15 and some browsers than 3 years old don't support it](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment#browser_compatibility): I would prefer to keep the old alternative, to not have to bump the engine for so little.
```suggestion
    aggregate[rank] = aggregate[rank] || []
```

Reply:
If it is used in JavaScript natively: yes. But TypeScript targets "es2016" currently and will output code compatible with es2016. This is while "esnext" (set as module) features are supported.

The JavaScript code for this ends up being this:

```js
var _a;
const aggregate = {};
for (const claim of propertyClaims) {
    const { rank } = claim;
    (_a = aggregate[rank]) !== null && _a !== void 0 ? _a : (aggregate[rank] = []);
    aggregate[rank].push(claim);
}
```

If this shouldn't also be in sourcecode the tsconfig should not allow for code newer than the target version. Otherwise I think its neat to use newer features and let TypeScript handle them to be supported in the specified target version.

Guess both is fine to me.

Reply:
Indeed. I would still prefer the version that does not need to be rewritten in JS, to keep the JS version readable.

Reply:
Interestingly TypeScript uses ES2016 currently and should not allow this operator.

Setting it to ES2015 for example also results in output which is not ES2015 compatible. (ES2016 introduces the `Array.prototype.includes` method which is even outputted on ES2015 settings.)

So I am a bit unsure why TypeScript seems to work differently here than I expected.
Ideally I would like to set the TypeScript config to reflect your view but I currently don't know how :D